src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,40 @@
 
 class Animal {
   // write your code here
+  constructor(name, health = 100, hidden = false) {
+    this.health = health;
+    this.name = name;
+    this.hidden = hidden;
+    Animal.alive.push(this);
+  }
+
+  static alive = [];
 }
 
 class Herbivore extends Animal {
   // write your code here
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  bite(animal) {
+    if (animal instanceof Carnivore) {
+      return;
+    }
+
+    if (animal.hidden === true) {
+      return;
+    }
+
+    animal.health -= 50;
```

Comment:
let's avoid using return statement here
let's just add the correct condition, and do some actions if condition = true